SpatialGDK/Documentation/content/porting-unreal-project-to-gdk.md

```diff
@@ -113,63 +128,68 @@ etc...
 
-1. Tag all replicated classes with the UCLASS tag `SpatialType`:
-    ``` 
-    UCLASS(SpatialType) 
-    ```
-1. Locate your game's `GameState`, and make it a public singleton:
+
+1. To provide context to SpatialOS on how to replicate [singleton classes]({{urlRoot}}/content/singleton-actors) (such as GameState and GameMode) we have added a `UCLASS` [specifier](https://docs.unrealengine.com/en-US/Programming/UnrealArchitecture/Reference/Classes/Specifiers) which provides said context. The next steps detail how to do this.
+   
+1. If your game's '`GameState`' is a cpp file, locate it and mark it as a public singleton by modifying the UCLASS specifier as shown:
     ```
     UCLASS(SpatialType=Singleton)
     ```
-1. Locate your game's `GameMode`, and make it a private singleton:
+    If your game's '`GameState`' is a blueprint then you will need to open the full blueprint editor for that GameState and navigate to the `Class Settings` in the blueprint editor toolbar. In the `Class Options` tick the `Spatial Type` checkbox and add the tag `Singleton` to the `Spatial Description` textbox.
+
+1. If your game's '`GameMode`' is a cpp file, locate it and mark it as a private singleton by modifying the UCLASS specifier as shown:
     ```
     UCLASS(SpatialType=(Singleton,ServerOnly))
     ```
+    If your game's '`GameMode`' is a blueprint then you will need to open the full blueprint editor for that GameMode and navigate to the `Class Settings` in the blueprint editor toolbar. In the `Class Options` tick the `Spatial Type` checkbox and add the tag `Singleton,ServerOnly` to the `Spatial Description` textbox.
+
+1. Marking these singleton classes as `Spatial Type` will cause spatial schema to be generated for said classes when pressing the `Schema` button in the `SpatialOSGDKToolbar`. We now need to mark the generated schema components as [streaming queries](https://docs.improbable.io/reference/13.3/shared/worker-configuration/bridge-config#streaming-queries) for your UnrealWorker worker configuration, this ensures your UnrealWorkers always have these singletons checked out. The generated schema is located at `<ProjectRoot>\spatial\schema\improbable\unreal\generated`.
+
+    Open the UnrealWorker worker config located in `<ProjectRoot>\spatial\workers\unreal\spatialos.UnrealWorker.worker.json` and add your new Singleton generated component(s) to the streaming queries.
     
-1. Open your UnrealWorker worker config located in `<ProjectRoot>\spatial\workers\unreal\spatialos.UnrealWorker.worker.json`. Add your new Singleton generated component(s) to the streaming queries.
-    
-For example:
+    For example:
```

Comment:
For example:
->
The `"streaming_query"` section of your configuration file should look something like this: